sdk/keyvault/keyvault-keys/test/public/list.spec.ts

```diff
@@ -9,239 +9,255 @@ import { assertThrowsAbortError } from "../utils/utils.common";
 import { testPollerProperties } from "../utils/recorderUtils";
 import { authenticate } from "../utils/testAuthentication";
 import TestClient from "../utils/testClient";
+import { supports, versionsToTest, SupportedVersions } from "@azure/test-utils-multi-version";
 
-describe("Keys client - list keys in various ways", () => {
-  const keyPrefix = `list${env.KEY_NAME || "KeyName"}`;
-  let keySuffix: string;
-  let client: KeyClient;
-  let testClient: TestClient;
-  let recorder: Recorder;
-
-  beforeEach(async function() {
-    const authentication = await authenticate(this);
-    keySuffix = authentication.keySuffix;
-    client = authentication.client;
-    testClient = authentication.testClient;
-    recorder = authentication.recorder;
-  });
-
-  afterEach(async function() {
-    await recorder.stop();
-  });
-
-  // The tests follow
-
-  // Use this while recording to make sure the target keyvault is clean.
-  // The next tests will produce a more consistent output.
-  // This test is only useful while developing locally.
-  it("can purge all keys", async function(): Promise<void> {
-    // WARNING: When TEST_MODE equals "record", all of the keys in the indicated KEYVAULT_NAME will be deleted as part of this test.
-    if (!isRecordMode()) {
-      return this.skip();
-    }
-    for await (const properties of client.listPropertiesOfKeys()) {
-      try {
-        await testClient.flushKey(properties.name);
-      } catch {
-        // Nothing to do here.
-      }
-    }
-    for await (const deletedKey of client.listDeletedKeys()) {
-      try {
-        await testClient.purgeKey(deletedKey.name);
-      } catch {
-        // Nothing to do here.
-      }
-    }
-  });
-
-  it("can get the versions of a key", async function() {
-    const keyName = testClient.formatName(`${keyPrefix}-${this!.test!.title}-${keySuffix}`);
-    await client.createKey(keyName, "RSA");
-    let totalVersions = 0;
-    for await (const version of client.listPropertiesOfKeyVersions(keyName)) {
-      assert.equal(
-        version.name,
-        keyName,
-        "Unexpected key name in result from listPropertiesOfKeyVersions()."
+versionsToTest(["7.0", "7.1"]).forEach((serviceVersion) => {
```

Comment:
it would be nice to not have to do a `forEach` loop around the entire test suite. Could we tweak it so each describes loops automatically over each version instead?

Reply:
I tried briefly but wasn't clear on how I would make the current service version to test available to enclosed `describe()` and `it()`.  I will have another look.

Reply:
Since you're already messing with the global scope, I presume you could create a global variable or symbol for it?

Reply:
No, the global scope is not modified. The `supports()` methods returns an wrapper object that has these methods with same names as Mocha global methods: `describe()`, `it()`, `describe.only()`, `it.only()`, etc., with modified behaviors.

I have attempted to use `Mocha.Context` to store the version but it didn't work as  I expected.

Reply:
I spent more than a day [trying to use `Mocha.Context`](https://github.com/jeremymeng/azure-sdk-for-js/tree/multi-version-attempt-2) but encountered various issues.  I now like the current approach more. Even there are some boilerplate code it's still quite simple.

Reply:
It's actually easy to move the `forEach` loop and `onVersion` definition into `versionsToTest()` https://github.com/Azure/azure-sdk-for-js/pull/12719/commits/118320dcc8e46752284efd3f9710e62bc7e9086b.